clustintime/visualization.py

```diff
@@ -0,0 +1,257 @@
+            Saving directory for the plot
+        prefix : str
+            Prefix for the image
+        task : dictionary or list, optional
+            Structure containing the times when the task is performed. The default is [].
+        repetition_time : float, optional
+            TR of the data. The default is 0.5.
+    
+        Returns
+        -------
+        None.
+    
+        """
+    
+        if self.tasks is None:
+            self.tasks = []
+    
+        plt.figure(figsize=[8, 8])
+        heatmatrix = np.zeros([int(self.labels.max()), len(self.labels)])
+        rownames = np.zeros([int(self.labels.max())]).astype(str)
+        x_values = np.linspace(0, len(self.labels) * self.repetition_time, len(self.labels)).astype(int)
+        file1d = pd.DataFrame()
+        for i in range(int(self.labels.max())):
+            selected_labels = np.array([0] * len(self.labels))
+            selected_labels[np.where(self.labels == i + 1)] = 1
+            file1d = pd.concat([file1d, pd.DataFrame(selected_labels)], axis=1)
+            heatmatrix[i] = selected_labels
+            rownames[i] = f"# {i+1}"
+    
+        heatmatrix = pd.DataFrame(heatmatrix, columns=x_values, index=rownames)
+        colors = sns.color_palette("Dark2", len(self.tasks) + 1)
+        sns.heatmap(heatmatrix, cmap="Greys", xticklabels=150, cbar=False)
+        plt.xlabel("Time in seconds", fontsize=10)
+        plt.ylabel("Clusters", fontsize=10)
+        legends = np.zeros([len(self.tasks)]).astype(str)
+        rectangles = [patches.Rectangle((0, 0), 1, 1, facecolor=colors[0])]
+        for idx, task in enumerate(self.tasks.values()):
+            plt.vlines(task / self.repetition_time, 0, self.labels.max(), linewidth=1.2, colors=colors[idx], alpha=0.5)
+            legends[idx] = f"task {idx}"
+            rectangles.append(patches.Rectangle((0, 0), 1, 1, facecolor=colors[idx + 1]))
+        plt.title(self.title)
+        plt.legend(
+            (rectangles),
+            np.array(legends),
+            bbox_to_anchor=[1, 0.5],
+            loc="center left",
+            handlelength=1,
+            handleheight=1,
+        )
+        plt.savefig(f"{self.saving_dir}/{self.prefix}_heatmap.png")
+        np.savetxt(f"{self.saving_dir}/{self.prefix}_heatmap.1D", file1d)
+    
+    
+    def show_table(self):
+        """
+        Table display of the results
+    
+        Parameters
+        ----------
+        labels : numpy array
+            Array of assigned clusters.
+    
+        Returns
+        -------
+        None.
+    
+        """
+        cluster, count = np.unique(self.labels, return_counts=True)
+        array = [cluster, count, count / len(self.labels)]
+        # display_table(np.transpose([["Cluster"], ["Count"], ["Percentage"]]))
+        # display_table(np.transpose(array))
+        table_result = pd.DataFrame({"Cluster": array[0], "Count": array[1], "Percentage": array[2]})
+        print(table_result)
```

Comment:
Using prints instead of loggers is an antipattern.
Consider using a logger for this.